requirements/RequirementsChecker.py

```diff
@@ -65,8 +68,16 @@ def detect_collisions(self, traffic_state:TrafficState):
 				vehicle_box = self.calculate_rectangular_bounding_box(vehicle)
 				if self.do_polygons_intersect(ego_box, vehicle_box):
 					CollisionWithVehicle(ego_vehicle.id, vid)
+		
+		for vid, pedestrian in self.pedestrians.items():
+			if pedestrian.sim_state not in [ActorSimState.ACTIVE, ActorSimState.ACTIVE.value]:
+				continue
+			
+			pedestrian_pos = [pedestrian.state.x, pedestrian.state.y]
 
-
+			if self.check_circle_rectangle_collision(pedestrian_pos, ego_box, pedestrian.PEDESTRIAN_RADIUS):
+				CollisionWithPedestrian(ego_vehicle.id, vid)
```

Comment:
`vid`? I see on line 71: `for vid, pedestrian`. I suggest using `pid` to avoid confusion.